thirdparty/CMakeLists.txt

```diff
@@ -1 +1,36 @@
-add_subdirectory(spdlog)
+include(FetchContent)
+
+###################################################################################################
+# - cnmem -----------------------------------------------------------------------------------------
+
+FetchContent_Declare(
+    cnmem
+    GIT_REPOSITORY https://github.com/NVIDIA/cnmem
+    GIT_TAG        37896cc9bfc6536a8c878a1e675835c22d827821
+    GIT_SHALLOW    true
+)
+
+FetchContent_GetProperties(cnmem)
+if(NOT cnmem_POPULATED)
+  FetchContent_Populate(cnmem)
+  set(CNMEM_INCLUDE_DIR "${cnmem_SOURCE_DIR}/include" PARENT_SCOPE)
+  set(CNMEM_SOURCE_DIR "${cnmem_SOURCE_DIR}/src" PARENT_SCOPE)
```

Comment:
These variables will only be set if `cnmem_POPULATED` is not True, which means that on a 2nd build (not tested in CI), we won't get `CNMEM_INCLUDE_DIR` or `CNMEM_SOURCE_DIR`. Suggest breaking these out of the if block here since the `GetProperties` call is what populates `cnmem_SOURCE_DIR`.

Reply:
`FetchContent_MakeAvailable` also imports the module, but cnmem's `CMakeLists.txt` is pretty old and causes errors. I moved the variables out of the if block.